src/source/StreamInfoProvider.c

```diff
@@ -49,31 +70,59 @@ STATUS freeStreamInfoProvider(PStreamInfo* ppStreamInfo)
     LEAVES();
     return retStatus;
 }
-
-// Creates H264 track info for video
-STATUS createH264VideoTrackInfo(PTrackInfo pTrackInfo)
+// Creates track info for video with given codec and sets the content type as per the codec
+STATUS createVideoTrackInfo(VIDEO_CODEC_ID videoCodecId, PCHAR contentType, PTrackInfo pTrackInfo)
```

Comment:
`NULL` check the args here as well.

Reply:
Added the `CHK(pTrackInfo != NULL && pStreamInfo->streamCaps.contentType != NULL, STATUS_NULL_ARG);` to `createAudioVideoStreamInfo` and `createVideoStreamInfo` which call this function

Reply:
Yes but this function is in the header file, and can be called by other functions we may develop in the future. It should have checks as well.
Add NULL checks here and in createAudioTrackInfo.

Reply:
Right, I added the NULL checks in `createAudioTrackInfo` and `createVideoTrackInfo` as well